example/linux/systemd/Talawa-api.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+# filepath: eg:  /home/purnendu/Development/talawa-api/example/linux/systemd/Talawa-api.sh 
+
+# Load nvm and use the correct Node.js version
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+nvm use 20.18.0
```

Comment:
for this one system node version should be same as talawa api node js version so adjust this acording to it

Reply:
instead of doing all this compare node js version from package.json and if it is equal then proceed further with next step else install that specific version then procced further

Reply:
LOG_FILE="/var/log/talawa-api.log"
TARGET_NODE_VERSION=$(jq -r '.engines.node' package.json 2>/dev/null)

# Ensure the log file exists
sudo touch $LOG_FILE    sometimes this error i am getting  /var/log/talawa-api.log: Permission denied

Reply:
# Ensure the log file exists
touch $LOG_FILE  here what we can do like first check if log file exist or not in the location if is is not then throw error that logfile is not present and tell first create log file then proceed furter and remove the touch command fully only check

Reply:
Error: Unable to read 'engines.node' from package.json.  now this is the error

Reply:
# Change to the project directory give your path
cd /home/purnendu/Development/talawa-api || exit 1

LOG_FILE="/var/log/talawa-api.log"
TARGET_NODE_VERSION=$(jq -r '.engines.node' package.json 2>/dev/null)  on first line i have make directory to run from home directory so make like package.json is being load from this directory this is way it should get package.json or else throw error package,json is not found in working directory

Reply:
# Attempt to read the required Node.js version from package.json
TARGET_NODE_VERSION=$(jq -r '.engines.node' package.json 2>/dev/null)   instead of this first find the package.json form working directory if package.json is not found then exit with error else find node version from it

Reply:
{
	"engines": { 
		"node": "22.12.0"
	},
	"homepage": "https://github.com/PalisadoesFoundation/talawa-api#readme",
}  this is structure of package.json how to find node version from it for targeted node js version

Reply:
No it is not installed if not installed then installd by checking it give echo also like jq command is required to find Node version from package.json then give echo installing jq then proceeding further

Reply:
don't install jq by sudo it will give further problem what can we do like first check if jq is present or not if not then give error jq is not  installed in system . It is required to parse node js version from package.json . So first install then try again. then exit . all this should be also in log file and in terminal it will be easy for user to understand